internal/cmd/config/config.go

```diff
@@ -337,7 +337,29 @@ func Parse(d string) (*Config, error) {
 		if !strutil.Printable(result.Worker.Name) {
 			return nil, errors.New("Worker name contains non-printable characters")
 		}
+
 		if result.Worker.TagsRaw != nil {
+			switch t := result.Worker.TagsRaw.(type) {
+			// We allow `tags` to be a simple string containing a reference to
+			// an environment variable (env://) / file path (file://).
+			// We then lookup whatever value is within and put it back
+			// into TagsRaw to get parsed.
+			case string:
+				rawTags, err := parseutil.ParsePath(t)
+				if err != nil && !errors.Is(err, parseutil.ErrNotAUrl) {
+					return nil, fmt.Errorf("Error parsing worker tags: %w", err)
+				}
+				result.Worker.TagsRaw = nil
+
+				err = hcl.Decode(result, rawTags)
```

Comment:
I wondering if this should decode into a new `*Config` instead of decoding again into `result`. If something other then tags is in the env/file referenced in the result of `parseutil.ParsePath(t)` won't this replace it? For example what happens if you add a test like:

```go
{
    name: "additional configuration in env",
    in: `
    worker {
        name = "dev-worker"
        tags = "env://BOUNDARY_WORKER_TAGS"
    }`,
    actualTags: `
    worker {
        name = "prod-worker"
        tags {
            type = ["dev", "local"]
        }
    }`,
    expWorkerTags: map[string][]string{
        "type": {"dev", "local"},
    },
    expErr: false,
},
```
And then add an assertion around `c.Worker.Name`?

Reply:
I knew it'd be replaced but I wasn't sure how to fix that problem - Can't believe I didn't see the way to do it right in front of me 😮‍💨 
This is a great suggestion, thank you!

Reply:
Fixed in dcb0944

Reply:
New and improved syntax 5675753